modules/vector-sets/vset.c

```diff
@@ -843,22 +843,45 @@ void VSIM_execute(RedisModuleCtx *ctx, struct vsetObject *vset,
         RedisModule_ReplyWithArray(ctx, REDISMODULE_POSTPONED_LEN);
     long long arraylen = 0;
 
+    int resp3 = RedisModule_GetContextFlags(ctx) & REDISMODULE_CTX_FLAGS_RESP3;
+
     for (unsigned int i = 0; i < found && i < count; i++) {
         if (distances[i] > epsilon) break;
         struct vsetNodeVal *nv = neighbors[i]->value;
         RedisModule_ReplyWithString(ctx, nv->item);
         arraylen++;
+
+        /* If the user asked for multiple properties at the same time using
+         * the RESP3 protocol, we wrap the value of the map into an N-items
+         * array. Two for now, since we have just two properties that can be
+         * requested.
+         *
+         * So in the case of RESP2 we will just have the flat reply:
+         * item, score, attribute. For RESP3 instead item -> [score, attribute]
+         */
+        if (resp3 && withscores && withattribs)
+            RedisModule_ReplyWithArray(ctx,2);
```

Comment:
Isn't it supposed to be a map rather than an array, based on the comment above? Also, why do it only for resp3 and not for resp2 as well?

Reply:
For RESP3, we create a map that is: map -> attribute_or_score if just one was selected, or map -> [score,attrib] if multiple will be selected. Basically the behavior will be that if just one extra information other than the element is requested, we return a simple map, otherwise we return a map mapping to an array. This is backward compatible, produces less protocol and so forth, but changes the reply format in case multiple additional items are requested. It is a tradeoff. The alternative would be that the case case would return item -> [score], a single item array, which is a bit odd.

Reply:
Got it, thanks